static/js/message_events.js

```diff
@@ -393,8 +393,10 @@ export function update_messages(events) {
             msg.raw_content = event.content;
         }
 
-        msg.last_edit_timestamp = event.edit_timestamp;
-        delete msg.last_edit_timestr;
+        if (!event.rendering_only) {
+            msg.last_edit_timestamp = event.edit_timestamp;
+            delete msg.last_edit_timestr;
```

Comment:
What does deleting `msg.last_edit_timestr` do?  I'm wondering if there's a latent bug here, since I don't see logic reading that field in a context where we didn't just recompute it.

Reply:
This pattern is echoed earlier in the code, for topic or stream edits (lines 247 & 248). So if there is an latent bug here, we'll need to address it in both places.

It looks like the set `last_edit_timestamp` and delete `last_edit_timestr` pattern here dates back to when the `message_events.js ` was created / extracted as a module ~5 yrs ago, which doesn't mean that the logic isn't off, but does mean it's been around for a while.

The three message fields that seem relevant here are:
1. `local_edit_timestamp` - set when an edit is sent to the server (`message_edit.js`)
2. `last_edit_timestamp` - set in the `update_messages` logic / function (`message_events.js`)
3. `last_edit_timestr` - set in `message_list_view.js` and that uses the above two values.

In `message_list_view.js`, all three are part of the function `_get_msg_timestring` (line 223), which is only called from the  `_add_msg_edited_vars` function (line 242). Here's the path(s) to get to that second function ...
- `_add_msg_edited_vars` is called by
  - `set_calculated_message_container_variables`, which is called by
    - `_rerender_message`, which is called by
      - `rerender_messages`, which is called in a few `js` modules
    - `build_message_groups`, which is called by
      - `render` in `message_list_view.js`

I haven't studied this area (building the message list view) of the front end code before now. Currently, it seems to me that the desire was to always have `last_edit_timestr` be undefined before it's calculated in `_add_msg_edited_vars` by `_get_msg_timestring`.

Reply:
As to why one would want the `last_edit_timestr` to be undefined for each message rerender call is what's not clear to me :thinking: